executor/executor.go

```diff
@@ -742,19 +744,13 @@ func (e *CheckTableExec) checkIndexHandle(ctx context.Context, src *IndexLookUpE
 	for {
 		err = Next(ctx, src, chk)
 		if err != nil {
+			e.retCh <- errors.Trace(err)
 			break
 		}
 		if chk.NumRows() == 0 {
 			break
 		}
-
-		select {
-		case <-e.exitCh:
-			return nil
-		default:
```

Comment:
if keeping it, we will have two placements to check whether it can exit or not and it will check the status too frequently. so I think it is more reasonable that it is removed.